go.mod

```diff
@@ -8,7 +8,10 @@ retract (
 	[v1.6.0, v1.6.1] // Contains data race bug in background connection establishment.
 )
 
+replace github.com/aws/aws-sdk-go => ./x/mongo/driver/auth/vendor/aws-sdk-go/
+
 require (
+	github.com/aws/aws-sdk-go v1.34.28
```

Comment:
I think copying is still reasonable for AWS SDK v1 because a direct import also brings dependencies such as `github.com/go-sql-driver/mysql`.

BTW, it is also the motivation for [GODRIVER-2731](https://jira.mongodb.org/browse/GODRIVER-2731) to import AWS SDK v2 when the driver officially supports golang 1.18.